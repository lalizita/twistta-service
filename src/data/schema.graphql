input CreateTweetInput {
  author: ID!
  parent: String
  text: String!
  clientMutationId: String
}

type CreateTweetPayload {
  tweets: [Tweet]
  error: String
  clientMutationId: String
}

type Mutation {
  UserRegister(input: RegisterUsernameInput!): RegisterUsernamePayload
  UserLogin(input: UserLoginInput!): UserLoginPayload
  CreateTweet(input: CreateTweetInput!): CreateTweetPayload
}

"""Query of users"""
type Query {
  users: [User]
  tweets(id: ID!): [Tweet]
}

input RegisterUsernameInput {
  username: String!
  email: String!
  password: String!
  clientMutationId: String
}

type RegisterUsernamePayload {
  username: String
  token: String
  clientMutationId: String
}

"""Tweets of a user"""
type Tweet {
  author: ID!
  parent: ID
  text: String
}

"""user data"""
type User {
  _id: String
  username: String!
  active: Boolean
  email: String!
}

input UserLoginInput {
  username: String!
  password: String!
  clientMutationId: String
}

type UserLoginPayload {
  token: String
  error: String
  clientMutationId: String
}
