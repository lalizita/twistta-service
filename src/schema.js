import { GraphQLSchema } from 'graphql';

import QueryType from './type/QueryType';
import MutationType from './type/MutationType';

console.log(MutationType)

export const schema = new GraphQLSchema({
  query: QueryType,
  mutation: MutationType
});