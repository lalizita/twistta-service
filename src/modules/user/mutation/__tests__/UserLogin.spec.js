import { graphql } from 'graphql'
import { connectMongoose, createUser, clearDatabase } from '../../../../../test/helper'
import { schema } from '../../../../schema'

const createUsers = async () => {
  await clearDatabase()
  await createUser('lary_joy', 'lary@mail.com', 'mydog123')
  await createUser('midorya', 'mydoria@mail.com', 'mydog123')
  await createUser('cat_joy', 'cast@mail.com', 'mydog123')
}

describe('User register test', () => {
  beforeAll(async () => {
    await connectMongoose()
    await createUsers()
  })
  it('Should not login with a invalid user', async () => {
    const query = `
      mutation M(
        $username: String!
        $password: String!
      ) {
        UserLogin(input: {
          username: $username
          password: $password
        }) {
          token
          error
        }
      }
    `
    const variables = {
      username: 'lalala',
      password: 'mydog123'
    }
    const {
      data: { UserLogin }
    } = await graphql(schema, query, {}, {}, variables)
    expect(UserLogin.error).toBe('invalid user')
    expect(UserLogin.token).toBe(null)
  })
  it('Should not login with invalid password', async () => {
    const query = `
        mutation M(
          $username: String!
          $password: String!
        ) {
          UserLogin(input: {
            username: $username
            password: $password
          }) {
            token
            error
          }
        }
      `
    const variables = {
      username: 'midorya',
      password: '123xxx'
    }
    const {
      data: { UserLogin }
    } = await graphql(schema, query, {}, {}, variables)
    expect(UserLogin.error).toBe('invalid password')
    expect(UserLogin.token).toBe(null)
  })
  it('Should not login correctly', async () => {
    const query = `
        mutation M(
          $username: String!
          $password: String!
        ) {
          UserLogin(input: {
            username: $username
            password: $password
          }) {
            token
            error
          }
        }
      `
    const variables = {
      username: 'lary_joy',
      password: 'mydog123'
    }
    const {
      data: { UserLogin }
    } = await graphql(schema, query, {}, {}, variables)
    expect(UserLogin.token).not.toBe(null)
    expect(UserLogin.error).toBe(null)
  })
})