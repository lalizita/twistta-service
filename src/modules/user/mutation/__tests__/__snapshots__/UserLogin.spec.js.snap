// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User register test Should make a login if the credentials are valid 1`] = `
Object {
  "username": "lais_registrada",
}
`;
